fedora_cloud_tests/azure.py

```diff
@@ -311,3 +445,36 @@ def _find_xml_file(self, log_path, run_name):
 
         _log.warning("No XML file with suffix 'lisa.junit.xml' found in %s", xml_path)
         return None
+
+    def _generate_ssh_key_pair(self, temp_dir):
+        """
+        Generate an SSH key pair for authentication.
+
+        Args:
+            temp_dir (str): Directory to store the generated key pair.
+
+        Returns:
+            str: Path to the private key file. or None if generation fails.
+        """
+
+        private_key_path = os.path.join(temp_dir, "id_rsa")
+        public_key_path = os.path.join(temp_dir, "id_rsa.pub")
+
+        try:
+            # Generate SSH key pair using ssh-keygen
+            cmd = ["ssh-keygen", "-t", "rsa", "-b", "2048", "-f", private_key_path, "-N", ""]
```

Comment:
These days ssh-keygen defaults to Ed25519 keys. Since Azure now supports using those (yay) I think switching this to 
```suggestion
            cmd = ["ssh-keygen", "-t", "ed25519", "-f", private_key_path, "-N", ""]
```

It's still a good idea to be explicit since if we just use the default Fedora may adopt a new key type before Azure supports it (sadly)